### PR TITLE
Adjust authenticator_input to authenticator_parameters conversion flow

### PR DESCRIPTION
**Pull request by sashaCher**
### What does this PR do?

Adjusts authenticator_input to authenticator_parameters conversion flow

#### Change log
- [ ] The CHANGELOG has been updated, or
- [X] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [X] This PR does not require updating any documentation

#### API Changes
- [ ] The [OpenAPI spec](https://github.com/cyberark/conjur-openapi-spec) has been updated to meet new API changes (or an issue has been opened), or
- [X] The changes in this PR do not affect the Conjur API

### CODE REVIEW

File: app/controllers/authenticate_controller.rb [hunk 1]
```diff
@@ -123,6 +123,18 @@ def authenticator_input
     )
   end
 
+  def authenticator_input_without_credentials
```
Comment (orenbm) [inline app/controllers/authenticate_controller.rb:126]: can you add a comment with the purpose of this method?

File: app/domain/authentication/authn_jwt/configuration_jwt_generic_vendor.rb [hunk 2]
```diff
@@ -18,8 +17,12 @@ def initialize
         )
       end
 
-      def extract_token_from_credentials(credentials)
-        @extract_token_from_credentials.call(credentials: credentials)
+      def create_authentication_parameters(authenticator_input)
+        @authentication_parameters = Authentication::AuthnJwt::AuthenticationParameters.new(authenticator_input)
+        @authentication_parameters.credentials = authenticator_input.request.read
```
Comment (orenbm) [inline app/domain/authentication/authn_jwt/configuration_jwt_generic_vendor.rb:22]: can this be done in the `AuthenticationParameters` init method?

File: app/domain/authentication/authn_jwt/configuration_jwt_generic_vendor.rb [hunk 3]
```diff
@@ -18,8 +17,12 @@ def initialize
...
+        @authentication_parameters.jwt_token = @extract_token_from_credentials.call(
```
Comment (orenbm) [inline app/domain/authentication/authn_jwt/configuration_jwt_generic_vendor.rb:23]: i will just note that there is a circular dependency here

Comment (orenbm) [inline app/domain/authentication/authn_jwt/configuration_jwt_generic_vendor.rb:23]: can this be done in the `AuthenticationParameters` init method?

Comment (orenbm) [reply app/domain/authentication/authn_jwt/configuration_jwt_generic_vendor.rb:23]: this piece of code indicates that something in the design is not tight

Comment (sashaCher) [reply app/domain/authentication/authn_jwt/configuration_jwt_generic_vendor.rb:23]: >this piece of code indicates that something in the design is not tight
@orenbm Can you please elaborate more

File: app/domain/authentication/authn_jwt/configuration_jwt_generic_vendor.rb (hunk 2 above)
Comment (sashaCher) [reply app/domain/authentication/authn_jwt/configuration_jwt_generic_vendor.rb:22]: Let's say that I would like to not couple `AuthenticationParameters` to request object
But I'll remove unnecessary credentials assignment.

File: app/controllers/authenticate_controller.rb (hunk 1 above)
Comment (sashaCher) [reply app/controllers/authenticate_controller.rb:129]: Done

File: app/domain/authentication/authn_jwt/configuration_jwt_generic_vendor.rb (hunk 3 above)
Comment (orenbm) [reply app/domain/authentication/authn_jwt/configuration_jwt_generic_vendor.rb:22]: you have an `authentication_parameters` object that has a `jwt_token` object that relies on the `authentication_parameters` object that contains it. it is a circular dependency.

I _think_ that this logic can be done inside `AuthenticationParameters` as both `credentials` and `jwt_token` need the `authenticator_input.request.read` which is already passed to the class

File: app/domain/authentication/authn_jwt/authentication_parameters.rb [hunk 4]
```diff
@@ -2,15 +2,14 @@ module Authentication
   module AuthnJwt
     # Data class to store data regarding jwt token that is needed during the jwt authentication process
     class AuthenticationParameters
-      attr_accessor :jwt_identity, :decoded_token, :jwt_token
-      attr_reader :authenticator_name, :service_id, :account, :username, :credentials, :client_ip, :request
+      attr_accessor :jwt_identity, :decoded_token, :jwt_token, :credentials
```
Comment (orenbm) [inline app/domain/authentication/authn_jwt/authentication_parameters.rb:5]: is ti possible to not have accessors? can (at least some of them) be initialized in the init method?

Comment (nessiLahav) [inline app/domain/authentication/authn_jwt/authentication_parameters.rb:5]: @sashaCher Are we using `credentials` in our flow ?

File: app/domain/authentication/authn_jwt/configuration_jwt_generic_vendor.rb (hunk 3 above)
Comment (nessiLahav) [reply app/domain/authentication/authn_jwt/configuration_jwt_generic_vendor.rb:22]: I dont have a strong opinion here, but if i must to choose, i prefer the POJO approach to leave the `AuthenticationParameters` class without external logic

File: app/domain/authentication/authn_jwt/authentication_parameters.rb (hunk 4 above)
Comment (sashaCher) [reply app/domain/authentication/authn_jwt/authentication_parameters.rb:5]: No we don't

**Review by nessiLahav — APPROVED** [review]
LGTM

**Pull request merged**